gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -50,6 +53,12 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                 return false;
             }
         });
+
+        // Check whether the activity was launched via a deeplink
```

Comment:
Am I interpreting the comment correct that uri is not null iff it was launched via a deeplink?

Reply:
Not in general (i.e. I know there could be cases where some other resource has an uri); it worked in this simple app but I have now considered changing it to first checking whether getIntent().getAction() equals Intent.ACTION_VIEW as I think it would be more accurate in more general cases